curvefs/proto/metaserver.proto

```diff
@@ -0,0 +1,181 @@
+/*
+ *  Copyright (c) 2020 NetEase Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+syntax="proto2";
+package curvefs.metaserver;
+option cc_generic_services = true;
+
+enum MetaStatusCode {
+    OK = 0;             // 成功
+    UNKNOWN_ERROR = 1;  // 未知错误
+    PARAM_ERROR = 2;
+}
+
+// dentry interface
+message GetDentryRequest {
+    required uint32 fsId = 1;
+    required uint64 parentInodeId = 2;
+    required string name = 3;
+}
+
+message Dentry {
+    required uint32 fsId = 1;
+    required uint64 inodeId = 2;
+    required uint64 parentInodeId = 3;
+    required string name = 4;
+}
+
+message GetDentryResponse {
+    required MetaStatusCode statusCode = 1;
+    optional Dentry dentry = 2;
+}
+
+message ListDentryRequest {
+    required uint32 fsId = 1;
+    required uint64 dirInodeId = 2;
+    optional uint32 last = 3;   // 从第几条开始
+    optional uint32 count = 4;    // 请求多少条数据
+}
+
+message ListDentryResponse {
+    required MetaStatusCode statusCode = 1;
+    repeated Dentry dentrys = 2;
+}
+
+message CreateDentryRequest {
+    required Dentry dentry = 1;
+}
+
+message CreateDentryResponse {
+    required MetaStatusCode statusCode = 1;
+}
+
+message UpdateDentryRequest {
+    required Dentry dentry = 1;
+}
+
+message UpdateDentryResponse {
+    required MetaStatusCode statusCode = 1;
+}
+
+message DeleteDentryRequest {
+    required uint32 fsId = 1;
+    required uint64 parentInodeId = 2;
+    required string name = 3;
+}
+
+message DeleteDentryResponse {
+    required MetaStatusCode statusCode = 1;
+}
+
+// inode interface
+message GetInodeRequest {
+    required uint32 fsId = 1;
+    required uint64 inodeId = 2;
+}
+
+enum FileType {
```

Comment:
conflict with 'FileType' in 'include/client/libcurve.h'

Reply:
chenge FileType to FsFileType